### PR TITLE
Integrate with the Backend

### PR DESCRIPTION
**Pull request by mechlin-sda-shahnawaaz**

### CODE REVIEW

File: src/hooks/useNavBar.tsx [hunk 1]
```diff
@@ -1,10 +1,9 @@
-
-import { useState, useEffect } from 'react';
-import { useLocation } from 'react-router-dom';
+import { useState, useEffect } from "react";
+import { useLocation } from "react-router-dom";
 
 const useNavbar = () => {
   const [isSidebarOpen, setIsSidebarOpen] = useState(false);
-  const [searchQuery, setSearchQuery] = useState('');
+  const [searchQuery, setSearchQuery] = useState("");
```
Comment (shivshankar12345) [inline src/hooks/useNavBar.tsx:6]: Why don't use type in useState ?

File: src/hooks/useNavBar.tsx [hunk 2]
```diff
@@ -17,7 +16,7 @@ const useNavbar = () => {
 
   const handleSearchSubmit = (e: React.FormEvent) => {
     e.preventDefault();
-    console.log('Search Query:', searchQuery);
+    console.log("Search Query:", searchQuery);
```
Comment (shivshankar12345) [inline src/hooks/useNavBar.tsx:19]: remove console

File: src/pages/HomePage.tsx [hunk 3]
```diff
@@ -1,45 +1,182 @@
...
+              <div
+                className="relative bg-cover bg-center h-[70vh] flex items-center justify-center"
+                style={{
+                  backgroundImage:
+                    'url("https://supertails.com/cdn/shop/files/12th_oct_web_2-min_1600x.png?v=1728541127")',
+                  width: "100%",
+                  //backgroundSize: "contain",
+                }}
+              ></div>
+            </SwiperSlide>
```
Comment (shivshankar12345) [inline src/pages/HomePage.tsx:48]: Make this carousel dynamic using map function

File: src/pages/HomePage.tsx [hunk 4]
```diff
@@ -1,45 +1,182 @@
...
+          </Swiper>
+        </section>
+
+        {/* Categories Section */}
+        <section className="py-16 bg-white">
+          <h2 className="text-3xl font-semibold text-center mb-8">
+            Explore Our Categories
+          </h2>
```
Comment (shivshankar12345) [inline src/pages/HomePage.tsx:82]: same for categories, make it dynamic, make a constant array of objects and add the required data in that then use map function to map through it.

File: src/pages/user/LogInPage.tsx [hunk 5]
```diff
@@ -7,38 +7,49 @@ import { baseURL } from "../../env";
...
+      // const resp = await axios.patch(
+      //   `${baseURL}/users/update`,
+      //   {
+      //     name: data.name,
+      //     phone: data.phone,
+      //     gender: data.gender,
+      //   },
+      //   { headers: { Authorization: `Bearer ${accessToken}` } }
+      // );
+      const resp = await makeAPICallWithData(
```
Comment (shivshankar12345) [inline src/pages/user/LogInPage.tsx:35]: remove commented codes

**Review by shivshankar12345 — CHANGES_REQUESTED** [review]

File: src/pages/HomePage.tsx (hunk 3 above)
Comment (mechlin-sda-shahnawaaz) [reply src/pages/HomePage.tsx:48]: dynamic crousel we will implement after the demo

File: src/pages/HomePage.tsx (hunk 4 above)
Comment (mechlin-sda-shahnawaaz) [reply src/pages/HomePage.tsx:82]: we will make category dynamic after the demo

**Review by shivshankar12345 — APPROVED** [review]

**Pull request merged**